compilerlibs/Makefile.compilerlibs

```diff
@@ -18,377 +18,346 @@
-  typing/typedecl_immediacy.cmo \
-  typing/typedecl_separability.cmo \
-  typing/typeopt.cmo \
-  typing/typedecl.cmo \
-  typing/rec_check.cmo \
-  typing/typecore.cmo \
-  typing/typeclass.cmo \
-  typing/typemod.cmo
-TYPING_CMI = \
-  typing/annot.cmi \
-  typing/outcometree.cmi
-
-LAMBDA = \
-  lambda/debuginfo.cmo \
-  lambda/lambda.cmo \
-  lambda/printlambda.cmo \
-  lambda/switch.cmo \
-  lambda/matching.cmo \
-  lambda/translobj.cmo \
-  lambda/translattribute.cmo \
-  lambda/translprim.cmo \
-  lambda/translcore.cmo \
-  lambda/translclass.cmo \
-  lambda/translmod.cmo \
-  lambda/tmc.cmo \
-  lambda/simplif.cmo \
-  lambda/runtimedef.cmo
-LAMBDA_CMI =
-
-COMP = \
-  bytecomp/meta.cmo \
-  bytecomp/opcodes.cmo \
-  bytecomp/bytesections.cmo \
-  bytecomp/dll.cmo \
-  bytecomp/symtable.cmo \
-  driver/pparse.cmo \
-  driver/compenv.cmo \
-  driver/main_args.cmo \
-  driver/compmisc.cmo \
-  driver/makedepend.cmo \
-  driver/compile_common.cmo
-COMP_CMI = \
-  file_formats/cmo_format.cmi \
-  file_formats/cmx_format.cmi \
-  file_formats/cmxs_format.cmi
+utils_SOURCES = $(addprefix utils/, \
+  config.mli config.ml \
+  build_path_prefix_map.mli build_path_prefix_map.ml \
+  misc.mli misc.ml \
+  identifiable.mli identifiable.ml \
+  numbers.mli numbers.ml \
+  arg_helper.mli arg_helper.ml \
+  local_store.mli local_store.ml \
+  load_path.mli load_path.ml \
+  clflags.mli clflags.ml \
+  profile.mli profile.ml \
+  terminfo.mli terminfo.ml \
+  ccomp.mli ccomp.ml \
+  warnings.mli warnings.ml \
+  consistbl.mli consistbl.ml \
+  strongly_connected_components.mli strongly_connected_components.ml \
+  targetint.mli targetint.ml \
+  int_replace_polymorphic_compare.mli int_replace_polymorphic_compare.ml \
+  domainstate.mli domainstate.ml \
+  binutils.mli binutils.ml \
+  lazy_backtrack.mli lazy_backtrack.ml \
+  diffing.mli diffing.ml \
+  diffing_with_keys.mli diffing_with_keys.ml)
+
+parsing_SOURCES = $(addprefix parsing/, \
+  location.mli location.ml \
+  unit_info.mli unit_info.ml \
+  asttypes.mli \
+  longident.mli longident.ml \
+  parsetree.mli \
+  docstrings.mli docstrings.ml \
+  syntaxerr.mli syntaxerr.ml \
+  ast_helper.mli ast_helper.ml \
+  camlinternalMenhirLib.mli camlinternalMenhirLib.ml \
+  parser.mly \
+  lexer.mll \
+  pprintast.mli pprintast.ml \
+  parse.mli parse.ml \
+  printast.mli printast.ml \
+  ast_mapper.mli ast_mapper.ml \
+  ast_iterator.mli ast_iterator.ml \
+  attr_helper.mli attr_helper.ml \
+  builtin_attributes.mli builtin_attributes.ml \
+  ast_invariants.mli ast_invariants.ml \
+  depend.mli depend.ml)
+
+typing_SOURCES = \
+$(addprefix typing/,\
+  annot.mli \
+  ident.mli ident.ml \
+  path.mli path.ml \
+  primitive.mli primitive.ml \
+  type_immediacy.mli type_immediacy.ml \
+  outcometree.mli \
+  shape.mli shape.ml \
+  types.mli types.ml \
+  btype.mli btype.ml \
+  oprint.mli oprint.ml \
+  subst.mli subst.ml \
+  predef.mli predef.ml \
+  datarepr.mli datarepr.ml) \
```

Comment:
For this list which has files in multiple places, I think it's better to have each entry with an explicit directory